keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -234,54 +234,77 @@ implementation.  The "Design Details" section below is for the real
 nitty-gritty.
 -->
 #### Terminology
+- **multicluster namespace** - A placeholder name for a group of
+  Kubernetes namespaces with the same name on different clusters.
+  This API specifies the semantics for sharing services
+  amongst the namespaces that belong to a multicluster namespace.
+
+  Implementations of this API are responsible for defining and tracking
+  multicluster namespaces and their membership. The specific mechanism
+  is out of scope of this proposal.
```

Comment:
I'm concerned about how this weakens assumptions around this API and puts more on the consumer to figure out how each implementation works, which doesn't help with the consistency we're going for. As we discussed in the weekly meeting, it's certainly possible to build a reasonable system that manages membership/scope for each namespace separately and there's value in that, but it's more challenging to reason about. It should be possible to use the MCS API with such system, but I don't think that should be the baseline.

Ideally, consumers can assume implementations follow namespace sameness across all clusters involved in MCS unless they've adopted an implementation that provides a more fine grained membership concept, in which case they can still use the MCS API but either they or the implementation will need to provide appropriate guidance/guardrails/feedback around implementation-specific behavior.